File: Algorithms/Easy/121_BestTimeToBuyAndSellStock/Solution.c
int maxProfit(int* prices, int pricesSize){
    int answer = 0;
    int buyingPrice = prices[0];
    
    for (int i = 1; i < pricesSize; i++){
        if (prices[i] < buyingPrice)
            buyingPrice = prices[i];
        else if (prices[i] - buyingPrice > answer)
            answer = prices[i] - buyingPrice;
    }
    
    return answer;
}

File: Algorithms/Easy/121_BestTimeToBuyAndSellStock/Solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        right = 1
        max_profit = 0

        while right < len(prices):
            if prices[right] < prices[left]:
                left = right
                right += 1
            else:
                max_profit = max(max_profit, prices[right] - prices[left])
                right += 1

        return max_profit